File: spades/Spades.Web/Hubs/GameHub.cs
using System.Linq;
using Spades.Models;

namespace Spades.Hubs
{
    public class GameHub : BaseHub
    {
        public void TakeSeat(User user, int seatId)
        {
            Game.Users[seatId] = Users.Single(u => u.ConnectionId == user.ConnectionId);
            Clients.All.TakeSeat(user, seatId);
        }

        public void RemoveFromSeat(int seatId)
        {
            Game.Users[seatId] = null;
            Clients.All.RemoveFromSeat(seatId);
        }
    }
}

File: spades/Spades.Web/Scripts/Index/gamehub.js
var gameHub = (function () {
    var self = this;
    self.server = $.connection.gameHub.server;
    self.client = $.connection.gameHub.client;

    self.client.takeSeat = function (user, seatId) {
        viewModel.game.takeSeat(getObservableUser(
                                  user.Username,
                                  user.Email,
                                  user.GravatarHash,
                                  user.ConnectionId), seatId);
    };

    self.client.removeFromSeat = function (seatId) {
        viewModel.game.removeFromSeat(seatId);
        //do some game pausing/freaking out... you know, once we get to that point
    };

    return {
        takeSeat: function (index) {
            self.server.takeSeat(ko.toJS(viewModel.user), index);
            viewModel.isPlaying(true);
        },
        removeFromSeat: function(index) {
            self.server.removeFromSeat(index);
            viewModel.isPlaying(false);
        }
    };
})();
